com/win32com/test/testPyComTest.py

```diff
@@ -866,6 +868,35 @@ def TestQueryInterface(long_lived_server=0, iterations=5):
         tester.TestQueryInterface()
 
 
+def TestRememberedGeneratedTypeLibraries():
+    typelib_ob = pythoncom.LoadTypeLib(
+        os.path.join(
+            localWin32comDirectory.parent, "TestSources", "PyCOMTest", "PyCOMTest.tlb"
+        )
+    )
+    # Make sure the TypeLib isn't already present, or that defeats the point of this test
+    assert typelib_ob not in gencache.demandGeneratedTypeLibraries.values()
+
+    module = gencache.EnsureModuleForTypelibInterface(typelib_ob, bForDemand=True)
+    tla = typelib_ob.GetLibAttr()
+    guid = tla[0]
+    lcid = tla[1]
+    major = tla[3]
+    minor = tla[4]
+    info = str(guid), lcid, major, minor
+    assert gencache.demandGeneratedTypeLibraries[info] is typelib_ob
+
+    # # TODO: How can I get a None module in EnsureModule to test versionRedirectMap ?
+    # old_is_readonly, gencache.is_readonly = gencache.is_readonly, True
+    # gencache.EnsureModule(*info)
+    # gencache.is_readonly = old_is_readonly
+    # assert gencache.versionRedirectMap[info] is module
```

Comment:
I'll need help with this.

Reply:
Updated version: I just do `gencache.versionRedirectMap[info] = module`, but a real code path that does it would be nice too.